longevity_tombstone_gc_test.py

```diff
@@ -0,0 +1,125 @@
+                    in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+        # for node in self.db_cluster.nodes:
+        #     node.run_nodetool(sub_cmd="repair", args=f"-pr {self.keyspace} {self.table}")
+
+        if wait_propagation_delay:
+            time.sleep(self.propagation_delay)
+        self.repair_date = datetime.datetime.now()
+        self.log.info('Run a major compaction for user-table on node')
+        self.db_node.run_nodetool("compact", args=f"{self.keyspace} {self.table}")
+        self.wait_no_compactions_running()
+
+    def test_switch_tombstone_gc_modes(self):
+        """
+        test steps:
+        -----------
+        Based on TWCS TTL 48h longevity configuration with a much shorter TTL (few minutes) and same for gc-grace-seconds.
+        Start with tombstone-gc-mode disabled
+        wait a duration of TTL.
+        count number of tombstones.
+        run a repair + a major compaction.
+        count again to see not fewer tombstones exit.
+        change gc-grace-seconds back to default of 10 days.
+        change gc-mode to repair
+        run a repair + wait propagation delay + run a major compaction.
+        verify no tombstones exist in post-repair-created sstables.
+        change gc-mode to immediate.
+        wait for load to end + TTL period.
+        Run a major compaction.
+        Verify no tombstones.
+        """
+        # pylint: disable=too-many-locals
+
+        self.create_tables_for_scylla_bench()
+        self.db_node = self.db_cluster.nodes[0]
+        self.run_post_prepare_cql_cmds()
+        stress_queue = []
+
+        stress_cmd = self.params.get('stress_cmd')
+        params = {'stress_cmd': stress_cmd, 'round_robin': self.params.get('round_robin')}
+        self._run_all_stress_cmds(stress_queue, params)
+
+        self.log.info('Wait a duration of TTL + propagation_delay_in_seconds')
+        wait_for_tombstones = 4 * 60 * 2
+        time.sleep(wait_for_tombstones)
+        self.db_node.run_nodetool(f"flush -- {self.keyspace}")
+        sstable_utils = SstableUtils(db_node=self.db_node, propagation_delay_in_seconds=self.propagation_delay,
+                                     ks_cf=self.ks_cf)
+        self.log.info('Count the initial number of tombstones')
+        tombstone_num_pre_repair = sstable_utils.count_tombstones()
+        self._run_repair_and_major_compaction()
+        tombstone_num_post_repair = sstable_utils.count_tombstones()
+        assert tombstone_num_post_repair >= tombstone_num_pre_repair, \
+            f"Found unexpected fewer tombstones: {tombstone_num_post_repair} / {tombstone_num_pre_repair}"
+
+        self.log.info("change gc-grace-seconds back to default of 10 days and tombstone-gc mode to 'repair'")
+        with self.db_cluster.cql_connection_patient(node=self.db_node) as session:
+            query = "ALTER TABLE scylla_bench.test with gc_grace_seconds = 36240 " \
```

Comment:
where did you come with the value for `gc_grace_seconds = 36240` ?

Reply:
i meant to set the default value of 10 days, but somehow missed it (although this value is good enough as well ~ 1 day).
The default should be 864000 as found here:
https://docs.scylladb.com/stable/cql/ddl.html

fixed anyway.